File: frontend/src/dto/google/GooglePlaceDto.ts
import type {PlaceType} from "../../enums/PlaceType.ts";
import type {GoogleReviewDto} from "./GoogleReviewDto.ts";

export interface GooglePlaceDto {
    placeType: PlaceType;
    name: string;
    description: string;
    rating: number;
    openingTime: string;
    closingTime: string;
    reviews: GoogleReviewDto[];
    latitude: number;
    longitude: number;
}

File: backend/src/main/java/com/example/backend/dto/google/GooglePlaceDTO.java
package com.example.backend.dto.google;

import com.example.backend.model.enums.PlaceType;
import lombok.*;
import lombok.experimental.FieldDefaults;

import java.time.LocalTime;
import java.util.List;

@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
@FieldDefaults(level = AccessLevel.PRIVATE)
public class GooglePlaceDTO {

    PlaceType placeType;

    String name;
    String description;
    Double rating;

    LocalTime openingTime;
    LocalTime closingTime;
    List<GoogleReviewDTO> reviews;
    Double latitude;
    Double longitude;
}